config/routes.rb

```diff
@@ -1,6 +1,10 @@
 Rails.application.routes.draw do
+  get '/users', to: 'users#index', as: 'users'
+  get '/users/:id', to: 'users#show', as: 'user'
+  get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
```

Comment:
- [OPTIONAL] I will encourage you to make use of resources for less code, you can achieve that with just three lines of code:
```
resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

Reply:
oh that's really nice Thank you so much @Olamarx